apps/system/js/cell_broadcast_system.js

```diff
@@ -70,13 +70,19 @@
       var serviceId = msg.serviceId || 0;
       var conn = window.navigator.mozMobileConnections[serviceId];
       var id = msg.messageId;
+      var cdmaCategory = msg.cdmaServiceCategory;
 
       // Early return CMAS messsage and let network alert app handle it. Please
       // ref http://www.etsi.org/deliver/etsi_ts/123000_123099/123041/
       // 11.06.00_60/ts_123041v110600p.pdf, chapter 9.4.1.2.2 Message identifier
-      // Message id from range 4370 to 4399(1112 hex to 112f hex) should be CMAS
-      // message and network alert will display detail information.
-      if (id >= 4370 && id < 4400) {
+      // for GSM and http://www.3gpp2.org/public_html/specs/
+      // C.R1001-G_v1.0_Param_Administration.pdf for CDMA.
+      // GSM Message id from range 4370 to 4399(1112 hex to 112f hex) and 
+      // CDMA service category from range 4096 to 4351(1000 hex to 10ff hex)
+      // should be CMAS and network alert will display detail information.
+
+      if ((id && (id >= 4370 && id < 4400)) ||
+          (cdmaCategory && (cdmaCategory >= 4096 && cdmaCategory <= 4351))) {
```

Comment:
To be more clear, in MozCellBroadcast.webidl [1],
"messageId" will always be available in webidl definition but, for CDMA message, this value is not specified. (Should be 0, I think)
"cdmaServiceCategory" is specified with "?" marked in web.idl, for GSM message, it will be specifically set to "null".
[1] https://dxr.mozilla.org/mozilla-central/source/dom/webidl/MozCellBroadcastMessage.webidl